tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_DuDt.cpp

```diff
@@ -8,13 +8,273 @@
 #include <random>
 
 #include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Identity.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/Mesh.hpp"
 #include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "Helpers/PointwiseFunctions/GeneralRelativity/TestHelpers.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Christoffel.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/IndexManipulation.hpp"
 #include "Utilities/Gsl.hpp"
 
 // IWYU pragma: no_forward_declare Tensor
 
 namespace {
+// Since repeatedly generating new numbers to compare to SpEC when the function
+// signature of the GH time derivative changes is difficult, we use a reference
+// implementation that we test against the Spectral Einstein Code (SpEC) to test
+// against our RHS changes.
+template <size_t Dim>
+std::tuple<tnsr::aa<DataVector, Dim, Frame::Inertial>,
+           tnsr::aa<DataVector, Dim, Frame::Inertial>,
+           tnsr::iaa<DataVector, Dim, Frame::Inertial>>
+gh_rhs_reference_impl(
```

Comment:
It could, though the benefit is not obvious to me. Where else other than testing the RHS would we use the reference implementation?

Reply:
I was thinking that it could be used for testing bjorhus form boundary conditions, which are mostly posed as `rhs + correction`; but looking closer - the large list of input tensors here makes this functions less useful for testing other code than perhaps simply calling the new `ComputeDuDt::apply`.

Reply:
Ah okay, that's fair. Yea, it's probably easiest (and generally best) to just use `ComputeDuDt::apply`. I've found the large number of input arguments to this function rather cumbersome (another reason why I think decreases the usage of compute tags actually simplifies the code a lot)

Reply:
sounds good!